### PR TITLE
Soar content addition to existing o365 solution

### PR DESCRIPTION
**Pull request by manishkumar1991**
Required items, please complete
   
   Change(s):
   - Addition of four new playbooks with one custom connector

   Reason for Change(s):
   - Soar content was not available earlier in o365 solution

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Solutions/Microsoft Defender for Office 365/Playbooks/O365_Defender_FunctionAppConnector/readme.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+This Functions App Connector is to connect Defender for office 365 API.
+
+### Authentication methods supported by this connector
+
+* Custom Authentication/Certificate Authentication
+
+### Prerequisites For Defender for office 365 API Functions App Connector
+
+* App Registration on azure active directory is needed , Beacuse this connector uses certificate thumprints for Authentication
```
Comment (rahul0216) [inline Solutions/Microsoft Defender for Office 365/Playbooks/O365_Defender_FunctionAppConnector/readme.md]: Correct the spelling 'Beacuse' and 'thumprints'.

Comment (manishkumar1991) [reply Solutions/Microsoft Defender for Office 365/Playbooks/O365_Defender_FunctionAppConnector/readme.md]: Done

**Review by rahul0216 — COMMENTED** [review]
Please address the inline comments.

File: Solutions/Microsoft Defender for Office 365/Playbooks/O365_Defender_FunctionAppConnector/readme.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+2. Once application is registered , Kindly follow the steps mentioned below
+	- Store the Application ID , that will be required later when deploying the playbooks.
+	- Create the self signed certificate : New-PnPAzureCertificate -OutPfx pnp.pfx -OutCert pnp.cer
+	- Upload your client certificate(cer file) to the AD application
+	- Upload your pfx file to Azure function : az webapp config ssl upload --certificate-file "e:\cert\pnp.pfx"  --name "<function app name>" --resource-group ""  --certificate-password "" --query thumbprint --output tsv
+	- Configure Function to allow the function to read the certificate : az functionapp config appsettings set --name <app-name> --resource-group <resource-group-name> --settings WEBSITE_LOAD_CERTIFICATES="comma-separated-certificate-thumbprints"
+    - Store the thumbprint in Keyvault under secrets and have the secret name handy ,that will be required to enter under Certificate_key_name during the playbook deployment 
+	- Set container size of function app : az functionapp update --name "Yourfunctionappname" --resource-group "Yourresourcegroupname" --set containerSize=1536
```
[the remark this answers is not in the document]
Comment (manishkumar1991) [reply Solutions/Microsoft Defender for Office 365/Playbooks/O365_Defender_FunctionAppConnector/readme.md:47]: I deployed an Azure function using ARM template and then tried to manually add an application setting related to certificate_read_entry in the portal.

Got the following error message:

Updating web app settings

The specified container size is not supported. Use one of these container sizes: 128, 192, 256, 320, 384, 448, 512, 576, 640, 704, 768, 832, 896, 960, 1024, 1088, 1152, 1216, 1280, 1344, 1408, 1472, 1536.

So due to this reason I have added that .

File: Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockMalwareFileExtension/readme.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+6. Select Scope - Resource group
+7. Select Subscription - where Playbook has been created
+8. Select Resource group - where Playbook has been created
+9. Select Role - Microsoft Sentinel Responder
+10. Click Save (It takes 3-5 minutes to show the added role.)
+#### d. Assign access policy on key vault for Playbook to fetch the secret key
+1. Select the Keyvault resource where you have stored the secret
+2. Click on Access policies Blade
+3. Click on Create
+4. Under Secret permissions columun , Select Get , List from "Secret Management Operations"
```
[the remark this answers is not in the document]
Comment (manishkumar1991) [reply Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockMalwareFileExtension/readme.md]: Done

File: Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockSender-EntityTrigger/readme.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+6. Select Scope - Resource group
+7. Select Subscription - where Playbook has been created
+8. Select Resource group - where Playbook has been created
+9. Select Role - Microsoft Sentinel Responder
+10. Click Save (It takes 3-5 minutes to show the added role.)
+#### d. Assign access policy on key vault for Playbook to fetch the secret key
+1. Select the Keyvault resource where you have stored the secret
+2. Click on Access policies Blade
+3. Click on Create
+4. Under Secret permissions columun , Select Get , List from "Secret Management Operations"
```
[the remark this answers is not in the document]
Comment (manishkumar1991) [reply Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockSender-EntityTrigger/readme.md]: Done

File: Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockSender/readme.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+6. Select Scope - Resource group
+7. Select Subscription - where Playbook has been created
+8. Select Resource group - where Playbook has been created
+9. Select Role - Microsoft Sentinel Responder
+10. Click Save (It takes 3-5 minutes to show the added role.)
+#### d. Assign access policy on key vault for Playbook to fetch the secret key
+1. Select the Keyvault resource where you have stored the secret
+2. Click on Access policies Blade
+3. Click on Create
+4. Under Secret permissions columun , Select Get , List from "Secret Management Operations"
```
[the remark this answers is not in the document]
Comment (manishkumar1991) [reply Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockSender/readme.md]: Done

File: Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockSpamDomain/readme.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+6. Select Scope - Resource group
+7. Select Subscription - where Playbook has been created
+8. Select Resource group - where Playbook has been created
+9. Select Role - Microsoft Sentinel Responder
+10. Click Save (It takes 3-5 minutes to show the added role.)
+#### d. Assign access policy on key vault for Playbook to fetch the secret key
+1. Select the Keyvault resource where you have stored the secret
+2. Click on Access policies Blade
+3. Click on Create
+4. Under Secret permissions columun , Select Get , List from "Secret Management Operations"
```
[the remark this answers is not in the document]
Comment (manishkumar1991) [reply Solutions/Microsoft Defender for Office 365/Playbooks/O365DefenderPlaybooks/o365-BlockSpamDomain/readme.md]: Done

**Review by rahul0216 — APPROVED** [review]
Changes look good.

**Pull request merged**